build-utils.js

```diff
@@ -94,41 +97,85 @@ function taskHarness(task, projectOrStar, ...args) {
  * also add a license header and create sourcemaps.
  *
  * @param  {Object} options Options object with 'projectDir' and 'rollupConfig'
- * @return {[type]}         [description]
+ * @return {Promise}
  */
 function buildJSBundle(options) {
-  const destPath = path.join(options.projectDir, 'build');
-  const licenseHeader = fs.readFileSync('LICENSE-HEADER', 'utf8');
+  return new Promise((resolve, reject) => {
+    const outputName = options.buildPath.replace(/^build\//, '');
+    const sources = [
+      `${options.projectDir}/src/**/*.js`,
+      '{lib,packages}/**/*.js',
+      'node_modules/*/**/*.js',
+    ];
 
-  if (options.outputName.indexOf('build/') !== 0) {
-    throw new Error('Expected options.ouputName to start with \'build/\'');
-  }
+    gulp.src(sources)
+      .pipe(sourcemaps.init())
+      .pipe(rollup(options.rollupConfig))
+      .pipe(gulpif(options.minify, babel({
```

Comment:
There's an optional Babel transpilation that happens depending on the `plugins` definition in `options.rollupConfig`.

And then, independent of that and after all the Rollup transformations happen, there's another Babel invocation, but with it just configured to minify via Babili.

It's kind of confusing that Babel in the second case is basically just a harness of kick off minification, but that's how it goes.